### PR TITLE
CIV-8624 Claimant Intent – Check your Answers Mediation

### PR DESCRIPTION
**Pull request by ElifDenizEsea**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-8624

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/services/features/claimantResponse/checkAnswers/buildFreeTelephoneMediationSection.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+const getContactName = (claim: Claim) => {
+  return claim.claimantResponse.mediation?.companyTelephoneNumber?.option === YesNo.NO ?
+    claim.claimantResponse.mediation.companyTelephoneNumber.mediationContactPerson :
+    claim.applicant1.partyDetails.contactPerson;
+};
+
+const getCanWeUse = (claim: Claim) => {
+  if (claim.claimantResponse.mediation?.canWeUse?.option) {
```
Comment (TatsyMaslova) [inline src/main/services/features/claimantResponse/checkAnswers/buildFreeTelephoneMediationSection.ts:36]: what happens if in this case option is YesNo.No? will it still return yes and is it what we want? in addition when you are returning a value you do not need an else statement

File: src/main/services/features/common/buildCommonHearingRequirements.ts [hunk 2]
```diff
@@ -1,6 +1,4 @@
-import {
-  addSupportRequiredList,
-} from 'services/features/response/checkAnswers/hearingRequirementsSection/addSupportRequiredList';
+
 import {summaryRow, SummaryRow} from 'models/summaryList/summaryList';
```
Comment (TatsyMaslova) [inline src/main/services/features/common/buildCommonHearingRequirements.ts:2]: I would be careful changing this as i believe this is used in release1?

File: src/main/services/features/claimantResponse/checkAnswers/buildFreeTelephoneMediationSection.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+const getMediationContactNumber = (companyTelephoneNumber : CompanyTelephoneNumber) => {
+  return companyTelephoneNumber.option === YesNo.YES ?
+    companyTelephoneNumber.mediationPhoneNumberConfirmation :
+    companyTelephoneNumber.mediationPhoneNumber;
+};
+
+const getContactName = (claim: Claim) => {
+  return claim.claimantResponse.mediation?.companyTelephoneNumber?.option === YesNo.NO ?
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/services/features/claimantResponse/checkAnswers/buildFreeTelephoneMediationSection.ts:30]: this is the place where claimant gives contact number in claimant response

File: src/main/services/features/common/buildCommonHearingRequirements.ts [hunk 4]
```diff
@@ -62,8 +62,8 @@ export const getSummaryRowForDisplayEvidenceYourself = (claim: Claim, claimId: s
   );
 };
 
-export const vulnerabilityQuestion = (claim: Claim, claimId: string, lng: string): SummaryRow => {
-  const option = claim.directionQuestionnaire?.vulnerabilityQuestions?.vulnerability?.option === YesNo.YES
+export const vulnerabilityQuestion = (claim: Claim, claimId: string, lng: string, directionQuestionnaire : DirectionQuestionnaire): SummaryRow => {
+  const option = directionQuestionnaire?.vulnerabilityQuestions?.vulnerability?.option === YesNo.YES
```
Comment (TatsyMaslova) [inline src/main/services/features/common/buildCommonHearingRequirements.ts:66]: why do we need an extra parameter with direction questionnaire if we can get direction questionnaire from claim?

File: src/main/services/features/common/buildCommonHearingRequirements.ts [hunk 5]
```diff
@@ -86,8 +86,8 @@ export const vulnerabilityInfo = (claim: Claim, claimId: string, lng: string): S
   );
 };
 
-export const giveEvidenceYourself = (claim: Claim, claimId: string, lng: string): SummaryRow => {
-  const option = claim.directionQuestionnaire?.defendantYourselfEvidence?.option === YesNo.YES
+export const giveEvidenceYourself = (claim: Claim, claimId: string, lng: string, directionQuestionnaire : DirectionQuestionnaire): SummaryRow => {
+  const option = directionQuestionnaire?.defendantYourselfEvidence?.option === YesNo.YES
```
Comment (TatsyMaslova) [inline src/main/services/features/common/buildCommonHearingRequirements.ts:90]: same comment as above

File: src/main/services/features/common/buildCommonHearingRequirements.ts [hunk 6]
```diff
@@ -163,8 +163,8 @@ export const getUnavailableDatesList = (claim: Claim, claimId: string, lng: stri
   );
 };
 
-export const getUnavailabilityReason = (claim: Claim, claimId: string, days: number, lng: string): SummaryRow => {
-  const whyUnavailableForHearing = claim.directionQuestionnaire.hearing.whyUnavailableForHearing?.reason;
+export const getUnavailabilityReason = (claim: Claim, claimId: string, days: number, lng: string, directionQuestionnaire : DirectionQuestionnaire): SummaryRow => {
```
Comment (TatsyMaslova) [inline src/main/services/features/common/buildCommonHearingRequirements.ts:166]: same comment as above

File: src/main/services/features/common/buildCommonHearingRequirements.ts (hunk 5 above)
Comment (ElifDenizEsea) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:90]: this is from claimant response questionnaire

File: src/main/services/features/common/buildCommonHearingRequirements.ts (hunk 6 above)
Comment (ElifDenizEsea) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:166]: this is from claimant response questionnaire

File: src/main/services/features/common/buildCommonHearingRequirements.ts (hunk 2 above)
Comment (ElifDenizEsea) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:2]: moved to another location, it is on line 14

File: src/main/services/features/common/buildCommonHearingRequirements.ts (hunk 4 above)
Comment (TatsyMaslova) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:66]: this is not an exceptionally elegant code with so many parameters. claimant response questionnaire also is inside claim right? in addition as mentioned above, if we are using this in release1 i would be extra careful changing it. if we are passing just direction questionnaire do we need claim then? have you tested whether your changes are working with release1?

Comment (TatsyMaslova) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:66]: I can still see claim that is not used in the parameter

File: src/main/services/features/claimantResponse/checkAnswers/buildFreeTelephoneMediationSection.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import {constructResponseUrlWithIdParams} from 'common/utils/urlFormatter';
+import {summarySection, SummarySection} from 'models/summaryList/summarySections';
+import {CAN_WE_USE_COMPANY_URL, CAN_WE_USE_URL, CITIZEN_FREE_TELEPHONE_MEDIATION_URL} from 'routes/urls';
+import {getLng} from 'common/utils/languageToggleUtils';
+import {summaryRow} from 'models/summaryList/summaryList';
+import {t} from 'i18next';
+
+const changeLabel = (lang: string | unknown): string => t('COMMON.BUTTONS.CHANGE', {lng: getLng(lang)});
+
+const getContactNumber = (claim: Claim) => {
```
Comment (TatsyMaslova) [inline src/main/services/features/claimantResponse/checkAnswers/buildFreeTelephoneMediationSection.ts:13]: still not inside the claim as we agreed during our discussion

File: src/main/services/features/common/buildCommonHearingRequirements.ts (hunk 6 above)
Comment (TatsyMaslova) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:165]: is this claim parameter used anywhere?

File: src/main/services/features/common/buildCommonHearingRequirements.ts [hunk 8]
```diff
@@ -193,8 +192,8 @@ export const getSpecificCourtLocation = (claim: Claim, claimId: string, lng: str
   );
 };
 
-export const displaySpecificCourtLocation = (claim: Claim, claimId: string, lng: string): SummaryRow => {
-  const hasSpecificCourtLocation = claim.directionQuestionnaire?.hearing?.specificCourtLocation?.courtLocation;
+export const displaySpecificCourtLocation = (claim: Claim, claimId: string, lng: string, directionQuestionnaire : DirectionQuestionnaire): SummaryRow => {
+  const hasSpecificCourtLocation = directionQuestionnaire?.hearing?.specificCourtLocation?.courtLocation;
```
Comment (TatsyMaslova) [inline src/main/services/features/common/buildCommonHearingRequirements.ts:196]: same thing here. is claim used here? i think this parameter is not used anymore

File: src/main/services/features/common/buildCommonHearingRequirements.ts (hunk 5 above)
Comment (TatsyMaslova) [reply src/main/services/features/common/buildCommonHearingRequirements.ts:89]: i think claim parameter is not used here anymore

**Review by TatsyMaslova — APPROVED** [review]

**Review by GorilaNaranja — APPROVED** [review]
I tested it and this PR includes the changes for my ticket https://tools.hmcts.net/jira/browse/CIV-8621

I think we tell QA team to test them together in your PR.

**Review by laperla11 — DISMISSED** [review]

**Review by GorilaNaranja — APPROVED** [review]

**Review by laperla11 — APPROVED** [review]

**Pull request closed without merging**